internal/controller/controlplane/helper.go

```diff
@@ -102,12 +103,45 @@ func (c *K0sController) createMachineFromTemplate(ctx context.Context, name stri
 		return nil, err
 	}
 
-	if err = c.Client.Patch(ctx, machineFromTemplate, client.Apply, &client.PatchOptions{
-		FieldManager: "k0smotron",
-	}); err != nil {
+	existingMachineFromTemplate := &unstructured.Unstructured{}
+	existingMachineFromTemplate.SetAPIVersion(machineFromTemplate.GetAPIVersion())
+	existingMachineFromTemplate.SetKind(machineFromTemplate.GetKind())
+	err = c.Get(ctx, client.ObjectKey{Namespace: machineFromTemplate.GetNamespace(), Name: machineFromTemplate.GetName()}, existingMachineFromTemplate)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			if err = c.Client.Patch(ctx, machineFromTemplate, client.Apply, &client.PatchOptions{
+				FieldManager: "k0smotron",
+			}); err != nil {
+				return nil, fmt.Errorf("error apply patching: %w", err)
+			}
+			return machineFromTemplate, nil
+		}
+
+		return nil, fmt.Errorf("error getting machine implementation: %w", err)
+	}
+
+	err = mergo.Merge(existingMachineFromTemplate, machineFromTemplate, mergo.WithSliceDeepCopy)
+	if err != nil {
 		return nil, err
 	}
 
+	spec, _, _ := unstructured.NestedMap(existingMachineFromTemplate.Object, "spec")
+	patch := unstructured.Unstructured{Object: map[string]interface{}{
+		"spec": spec,
+	}}
+	data, err := patch.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+
+	resourceType := strings.ToLower(machineFromTemplate.GetKind()) + "s"
```

Comment:
wdyt about find the apiresource.Name using resource kind and rely on the crd plural?

Reply:
Changed it to more reliable way